pkg/util/kernel/proc.go

```diff
@@ -49,3 +55,50 @@ func WithAllProcs(procRoot string, fn func(int) error) error {
 	}
 	return nil
 }
+
+// scanNullString is a SplitFunc for a Scanner that returns each null-terminated string as a token
```

Comment:
add comments about each param. the second one is not clear just judging by its name

Reply:
also document the return values